miden-remote-provers/src/prover.rs

```diff
@@ -77,20 +77,19 @@ impl TransactionProver for RemoteTransactionProver {
             TransactionProverError::other_with_source("failed to connect to the remote prover", err)
         })?;
 
-        let mut client = self.client.write();
+        let mut client = self
+            .client
+            .write()
+            .take()
+            .ok_or_else(|| TransactionProverError::other("client should be connected"))?;
```

Comment:
Question: why do we need to do `take()` here now? (I don't think this was needed before, right?)